social-network-api/src/main/java/fs/socialnetworkapi/service/PasswordResetService.java

```diff
@@ -2,21 +2,17 @@
 
 import fs.socialnetworkapi.component.AppLink;
 import fs.socialnetworkapi.entity.User;
-import lombok.RequiredArgsConstructor;
-import org.springframework.beans.factory.annotation.Autowired;
+import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Service;
 
 import java.util.Optional;
 import java.util.UUID;
 
 @Service
-@RequiredArgsConstructor
+@AllArgsConstructor
 public class PasswordResetService {
-  @Autowired
   private final UserService userService;
-  @Autowired
   private final MailService mailService;
-  @Autowired
   private final AppLink appLink;
```

Comment:
We don't need a separate class for AppLink. It's injected using the annotation from the application.yml file. You can use it with easy action 
```
@Value("{myapp.baseUrl}")
private String applicationUrl;
```